hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -896,8 +899,10 @@ private AbfsRestOperation conditionalCreateOverwriteFile(final String relativePa
 
         try {
           // overwrite only if eTag matches with the file properties fetched before.
-          if (getPrefixMode() == PrefixMode.BLOB) {
-            boolean isNormalBlob = !checkIsBlobOrMarker(metadata);
+          boolean isNormalBlob = !checkIsBlobOrMarker(metadata);
+          if (!isNormalBlob && !abfsConfiguration.shouldMkdirFallbackToDfs() && getPrefixMode() == PrefixMode.BLOB) {
```

Comment:
This will be hit if shouldMkdirFallbackToDfs is not enabled only.

To make the flow more simpler, as discussed lets have createpath have logic as :
if (isMarker) { ..} if within the logic prefixMode is DFS or shouldMkdirFallback is true, then its an invalid state to be in, throw exception. else proceed with createpathblob.
in the exception block, add in comments that marker blob creations are not possible with DFS APIs (DFS rejects metadata hdi_isfolder)